src/charm.py

```diff
@@ -228,11 +225,19 @@ def _on_collect_unit_status(self, event: ops.CollectStatusEvent) -> None:
 
         if not snap_service.present:
             event.add_status(
-                BlockedStatus("snap service is not installed, please check snap service")
+                BlockedStatus(
+                    f"{SNAP_NAME} snap is not installed. "
+                    "Please wait for installation to complete, "
+                    "or manually reinstall the snap if the issue persists."
+                )
```

Comment:
I'm a bit confused now. Don't we need to add `return` right after "add_status"?

I initially thought the framework would select the highest-priority status across all units. However, for an individual unit, it seems more logical to return only one specific status. Returning blocked + active statuses at the same time feels odd to me. I'm unsure what the correct approach is, but I guess the final outcome would be the same.  

Here is the example code in document. 
https://ops.readthedocs.io/en/latest/#ops.CollectStatusEvent
```
    def _on_collect_status(self, event: ops.CollectStatusEvent):
        if 'port' not in self.model.config:
            event.add_status(ops.BlockedStatus('please set "port" config'))
            return
        event.add_status(ops.ActiveStatus())
```

Please feel free to close this conversation if the final outcome is the same. I just want to double check . thanks!